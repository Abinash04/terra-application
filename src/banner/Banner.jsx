import React from 'react';
import PropTypes from 'prop-types';
import uuidv4 from 'uuid/v4';

import BannerRegistrationContext from './_BannerRegistrationContext';
import { BANNER_TYPES } from './_Utils';

const propTypes = {
  /**
   * An action element to be added to the action section of the banner.
   */
  action: PropTypes.element,
  /**
   * Nodes providing the message content for the banner. Can contain text and HTML.
   */
  description: PropTypes.node,
  /**
   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause
   * the Dismiss button to be included on the banner.
   */
  onDismiss: PropTypes.func,
  /**
   * The title for the banner which will be bolded. Providing a title will override the default title.
   */
  title: PropTypes.string,
  /**
   * The type of alert to be rendered. One of `alert`, `error`, `warning`, `info`, `success` or `custom`.
   */
  type: PropTypes.oneOf([
    BANNER_TYPES.ALERT,
    BANNER_TYPES.ERROR,
    BANNER_TYPES.WARNING,
    BANNER_TYPES.INFO,
    BANNER_TYPES.SUCCESS,
    BANNER_TYPES.CUSTOM,
  ]).isRequired,
};

const Banner = (props) => {
  /**
   * A unique identifier is generated for each Banner during construction. This will be used to
   * uniquely register/unregister the banner with ancestor Banner Managers without requiring consumers to
   * define unique identifiers themselves.
   */
  const uuid = React.useRef(uuidv4());
  const bannerRegistration = React.useContext(BannerRegistrationContext);

  React.useEffect(() => {
    /**
     * If the bannerRegistration value is the ProviderRegistrationContext's default value,
     * then there is not a matching BannerCheckpoint above it in the hierarchy.
     * This is possible but likely not intentional, so the component warns.
     */
    if (!bannerRegistration && process.env.NODE_ENV !== 'production') {
      /* eslint-disable no-console */
      console.warn('A Banner was not rendered within the context of a BannerCheckpoint. If this is unexpected, validate that the expected version of the terra-application package is installed.');
      /* eslint-enable no-console */
    }

    bannerRegistration.registerBanner(uuid.current, props);

    return () => {
      bannerRegistration.unregisterBanner(uuid.current, props.type);
    };
  }, [props]);

  return null;
};

Banner.propTypes = propTypes;

export default Banner;
