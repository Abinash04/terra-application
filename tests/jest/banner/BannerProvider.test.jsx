import React, { useState } from 'react';
import { act } from 'react-dom/test-utils';
import { render, fireEvent } from '@testing-library/react';
import 'terra-base';
import { IntlProvider } from 'react-intl';

import BannerRegistrationContext from '../../../src/banner/private/BannerRegistrationContext';
import Banner, { BannerProvider } from '../../../src/banner';
// eslint-disable-next-line import/no-unresolved, import/extensions
import { messages } from '../../../aggregated-translations/en'; // aggregation is pre-jest step so this will exist

jest.mock('terra-base');
global.console.warn = jest.fn();

let keyValue = 0;
const mockBannerProps = { type: 'error', 'data-testid': 'rendered-banner' };

// eslint-disable-next-line react/prop-types
const ChildContent = ({ showBannerOnRender = false, buttonId = 'show-banner', children = null }) => {
  const [hasBanner, setHasBanner] = useState(showBannerOnRender);
  keyValue += 1;
  return (
    <div>
      <button
        type="button"
        data-testid={buttonId}
        onClick={() => setHasBanner(!hasBanner)}
      >
        Toggle Banner
      </button>
      {hasBanner ? <Banner {...mockBannerProps} key={keyValue} /> : null}
      {children}
    </div>
  );
};

const renderComponentWithChild = (childrenContent, checkpointProps = {}) => {
  let childContext;

  const component = render(
    <IntlProvider locale="en" messages={messages}>
      <BannerProvider {...checkpointProps}>
        <BannerRegistrationContext.Consumer>
          {(context) => {
            childContext = context;
            return childrenContent;
          }}
        </BannerRegistrationContext.Consumer>
      </BannerProvider>
    </IntlProvider>,
  );

  return { component, context: childContext };
};

describe('BannerProvider', () => {
  beforeEach(() => {
    jest.resetAllMocks;
    // eslint-disable-next-line no-console
    console.warn.mockClear();
  });

  it('renders with children', () => {
    const { component } = renderComponentWithChild(null);
    expect(component.container).toMatchSnapshot();
  });

  it('renders with children', () => {
    const { component } = renderComponentWithChild(<div>child</div>);

    expect(component.container).toMatchSnapshot();
  });

  it('renders with children that has a an banner', () => {
    const { component } = renderComponentWithChild(<ChildContent showBannerOnRender />);

    expect(component.container).toMatchSnapshot();
  });

  describe('registerBanner', () => {
    it('provides registerBanner callback in context', () => {
      const { context } = renderComponentWithChild();

      expect(context).toBeDefined();
      expect(context).toHaveProperty('registerBanner', expect.any(Function));
    });

    it('does not register banner if no id is provided', () => {
      const { component, context } = renderComponentWithChild();
      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(0);

      context.registerBanner(undefined, mockBannerProps);

      // eslint-disable-next-line no-console
      expect(console.warn).toHaveBeenCalledWith('A banner cannot be registered without an identifier.');
      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(0);
    });

    it('registers banner when id is provided', () => {
      const { component, context } = renderComponentWithChild();
      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(0);

      act(() => {
        context.registerBanner('mockID', mockBannerProps);
      });

      // eslint-disable-next-line no-console
      expect(console.warn).not.toHaveBeenCalledWith('A banner cannot be registered without an identifier.');
      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);
    });

    it('registers banner when rendered after mount', () => {
      const { component } = renderComponentWithChild(<ChildContent />);

      expect(component.container).toMatchSnapshot();

      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);
      expect(component.container).toMatchSnapshot();
    });
  });

  describe('unregistersBanner', () => {
    it('provides unregisterBanner callback in context', () => {
      const { context } = renderComponentWithChild();

      expect(context).toBeDefined();
      expect(context).toHaveProperty('unregisterBanner', expect.any(Function));
    });

    it('does not unregister banner if no id is provided', () => {
      const { component, context } = renderComponentWithChild(<ChildContent showBannerOnRender />);

      // eslint-disable-next-line no-console
      console.warn.mockClear(); // clear terra-responsive-element will be deprecated warning

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);

      context.unregisterBanner(undefined);

      // eslint-disable-next-line no-console
      expect(console.warn).toHaveBeenCalledWith('A banner cannot be unregistered without an identifier or banner type.');
      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);
    });

    it('does not unregister banner if banner type is not provided', () => {
      const { component, context } = renderComponentWithChild(<ChildContent showBannerOnRender />);

      // eslint-disable-next-line no-console
      console.warn.mockClear(); // clear terra-responsive-element will be deprecated warning

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);

      context.unregisterBanner('mockID', undefined);

      // eslint-disable-next-line no-console
      expect(console.warn).toHaveBeenCalledWith('A banner cannot be unregistered without an identifier or banner type.');
      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);
    });

    it('unregisters banner when id is provided', () => {
      const { component } = renderComponentWithChild(<ChildContent showBannerOnRender />);
      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);

      expect(component.container).toMatchSnapshot();

      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(0);
      expect(component.container).toMatchSnapshot();
    });
  });

  describe(('Nested Banner Checkpoint'), () => {
    it('registers banner with top-level Banner Checkpoint', () => {
      const childComponent = (
        <ChildContent>
          <BannerProvider>
            <ChildContent buttonId="show-2nd-banner" />
          </BannerProvider>
        </ChildContent>
      );
      const { component } = renderComponentWithChild(childComponent);
      expect(component.container).toMatchSnapshot();

      // trigger top checkpoint's banner
      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);
      expect(component.container).toMatchSnapshot();

      // trigger nested banner checkpoint's banner
      const secondChildButton = component.getByTestId('show-2nd-banner');
      act(() => {
        fireEvent.click(secondChildButton);
      });

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(2);
      expect(component.container).toMatchSnapshot();
    });

    it('unregisters banner with top-level Banner Checkpoint', () => {
      const childComponent = (
        <ChildContent showBannerOnRender>
          <BannerProvider>
            <ChildContent buttonId="show-2nd-banner" showBannerOnRender />
          </BannerProvider>
        </ChildContent>
      );
      const { component } = renderComponentWithChild(childComponent);

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(2);

      expect(component.container).toMatchSnapshot();

      // remove top checkpoint's banner
      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(1);
      expect(component.container).toMatchSnapshot();

      // remove nested banner checkpoint's banner
      const secondChildButton = component.getByTestId('show-2nd-banner');
      act(() => {
        fireEvent.click(secondChildButton);
      });

      expect(component.queryAllByTestId('rendered-banner')).toHaveLength(0);
      expect(component.container).toMatchSnapshot();
    });
  });
});
