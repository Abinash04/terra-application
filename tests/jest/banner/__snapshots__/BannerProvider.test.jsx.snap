// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerProvider Nested Banner Checkpoint registers banner with top-level Banner Checkpoint 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            />
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider Nested Banner Checkpoint registers banner with top-level Banner Checkpoint 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-testid="rendered-banner"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            />
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider Nested Banner Checkpoint registers banner with top-level Banner Checkpoint 3`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-testid="rendered-banner"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            >
              <div
                class="alert-base error narrow"
                data-testid="rendered-banner"
                role="alert"
              >
                <div
                  class="body body-std"
                >
                  <span
                    class="icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="tui-Icon icon IconError"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 48 48"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                        fill="#E50000"
                      />
                      <path
                        d="M7 21.5h34v5H7v-5z"
                        fill="#FFF"
                      />
                    </svg>
                  </span>
                  <div
                    class="section"
                  >
                    <strong
                      class="title"
                    >
                      <span>
                        Error.
                      </span>
                    </strong>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider Nested Banner Checkpoint unregisters banner with top-level Banner Checkpoint 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-testid="rendered-banner"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            >
              <div
                class="alert-base error narrow"
                data-testid="rendered-banner"
                role="alert"
              >
                <div
                  class="body body-std"
                >
                  <span
                    class="icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="tui-Icon icon IconError"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 48 48"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                        fill="#E50000"
                      />
                      <path
                        d="M7 21.5h34v5H7v-5z"
                        fill="#FFF"
                      />
                    </svg>
                  </span>
                  <div
                    class="section"
                  >
                    <strong
                      class="title"
                    >
                      <span>
                        Error.
                      </span>
                    </strong>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider Nested Banner Checkpoint unregisters banner with top-level Banner Checkpoint 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            >
              <div
                class="alert-base error narrow"
                data-testid="rendered-banner"
                role="alert"
              >
                <div
                  class="body body-std"
                >
                  <span
                    class="icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="tui-Icon icon IconError"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 48 48"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                        fill="#E50000"
                      />
                      <path
                        d="M7 21.5h34v5H7v-5z"
                        fill="#FFF"
                      />
                    </svg>
                  </span>
                  <div
                    class="section"
                  >
                    <strong
                      class="title"
                    >
                      <span>
                        Error.
                      </span>
                    </strong>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider Nested Banner Checkpoint unregisters banner with top-level Banner Checkpoint 3`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            />
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider registerBanner registers banner when rendered after mount 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider registerBanner registers banner when rendered after mount 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-testid="rendered-banner"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider renders with children 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      />
    </div>
  </div>
</div>
`;

exports[`BannerProvider renders with children 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          child
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider renders with children that has a an banner 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider unregistersBanner unregisters banner when id is provided 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-testid="rendered-banner"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BannerProvider unregistersBanner unregisters banner when id is provided 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
